packages/react/src/components/CodeSnippet/__tests__/CodeSnippet-test.js

```diff
@@ -6,102 +6,186 @@
-      );
-      clickWrapper.find(Copy).simulate('click');
-      expect(onClick).toHaveBeenCalled();
-    });
+  it('should allow hiding the copy button', () => {
+    render(
+      <CodeSnippet type="single" hideCopyButton>
+        {single}
+      </CodeSnippet>
+    );
+
+    expect(document.querySelector('button')).not.toBeInTheDocument();
   });
 
-  describe('check for showMoreBtn', () => {
-    let wrapper;
+  it('should set disabled on copy button if it is passed via props', () => {
+    render(
+      <CodeSnippet type="single" disabled>
+        {single}
+      </CodeSnippet>
+    );
+
+    expect(screen.getByTitle('Copy to clipboard')).toHaveAttribute('disabled');
+  });
+});
+
+describe('CodeSnippet events', () => {
+  it('should call the click handler when the copy button is clicked', () => {
+    const onClick = jest.fn();
+    render(
+      <CodeSnippet type="single" onClick={onClick}>
+        {single}
+      </CodeSnippet>
+    );
+
+    const button = screen.getByTitle('Copy to clipboard');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
 
-    beforeEach(() => {
-      wrapper = mount(<CodeSnippet type={'multi'} />);
-    });
+  it('should call the click handler with type inline', () => {
+    const onClick = jest.fn();
+    render(
+      <CodeSnippet type="inline" data-testid="code-6" onClick={onClick}>
+        {inline}
+      </CodeSnippet>
+    );
+
+    const button = screen.getByTestId('code-6');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+});
 
-    it('when less then 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14',
-      });
+describe('Show more button', () => {
+  it('should not have show more button when less then 15 rows', () => {
+    render(<CodeSnippet type="multi">{multiShort}</CodeSnippet>);
 
-      expect(wrapper.find(Button).length).toBe(0);
-    });
+    expect(
+      document.querySelector('button.cds--snippet-btn--expand')
+    ).not.toBeInTheDocument();
+  });
 
-    it('when exactly 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15',
-      });
+  it('should not have show more button when exactly 15 rows', () => {
+    render(<CodeSnippet type="multi">{multi15}</CodeSnippet>);
 
-      expect(wrapper.find(Button).length).toBe(0);
-    });
+    expect(
+      document.querySelector('button.cds--snippet-btn--expand')
+    ).not.toBeInTheDocument();
+  });
 
-    it.skip('when more then 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16',
-      });
+  it.skip('should have show more button when more than 15 rows', () => {
+    render(<CodeSnippet type="multi">{multiLong}</CodeSnippet>);
 
-      expect(wrapper.find(Button).length).toBe(1);
-    });
+    expect(
+      document.querySelector('button.cds--snippet-btn--expand')
```

Comment:
Instead of the querySelector, could this (and the others in lines above) be `screen.getByTitle('Show more');` or `screen.getByText('Show more');`? Would it maybe fix the issue of the test not passing?

Reply:
@tay1orjones So I tried using that for the first 2 tests but it fails bc `show more` isn't in the document.
![Screen Shot 2022-06-22 at 2 53 06 PM](https://user-images.githubusercontent.com/32556167/175124696-b81bfd62-5d08-4333-8bce-2494125e5450.png)


I also tried using that for the last test that I skipped, but even with that selector, it fails. It seems to not show up at all, even if I manually add `maxCollapsedNumberOfRows={15}`.